--- origsrc/lzlib-1.1/Makefile.in	2011-01-03 07:29:50.000000000 -0600
+++ src/lzlib-1.1/Makefile.in	2011-03-28 19:06:04.178031100 -0500
@@ -22,13 +22,13 @@ all : $(progname) $(progname_shared)
 lib$(libname).a: $(lib_objs)
 	$(AR) -rcs $@ $^
 
-lib$(libname).so.$(pkgversion) : $(sh_lib_objs)
-	$(CXX) -shared -Wl,--soname=lib$(libname).so.$(soversion) -o $@ $^
+lib$(libname).dll.a : $(sh_lib_objs)
+	$(CXX) -shared -Wl,--out-implib,$@ -o cyg$(libname)-$(soversion).dll $^
 
 $(progname) : $(objs) lib$(libname).a
 	$(CXX) $(LDFLAGS) -o $@ $^
 
-$(progname)_shared : $(objs) lib$(libname).so.$(pkgversion)
+$(progname)_shared : $(objs) lib$(libname).dll.a
 	$(CXX) $(LDFLAGS) -o $@ $^
 
 $(progname)_profiled : $(objs) lib$(libname).a
@@ -92,16 +92,12 @@ check : all bbexample lzcheck
 install : all install-info
 	if [ ! -d "$(DESTDIR)$(includedir)" ] ; then $(INSTALL_DIR) "$(DESTDIR)$(includedir)" ; fi
 	if [ ! -d "$(DESTDIR)$(libdir)" ] ; then $(INSTALL_DIR) "$(DESTDIR)$(libdir)" ; fi
+	if [ ! -d "$(DESTDIR)$(bindir)" ] ; then $(INSTALL_DIR) "$(DESTDIR)$(bindir)" ; fi
 	$(INSTALL_DATA) $(VPATH)/$(libname)lib.h "$(DESTDIR)$(includedir)/$(libname)lib.h"
 	$(INSTALL_DATA) ./lib$(libname).a "$(DESTDIR)$(libdir)/lib$(libname).a"
 	if [ -n "$(progname_shared)" ] ; then \
-	  $(INSTALL_PROGRAM) ./lib$(libname).so.$(pkgversion) "$(DESTDIR)$(libdir)/lib$(libname).so.$(pkgversion)" ; \
-	  if [ -e "$(DESTDIR)$(libdir)/lib$(libname).so.$(soversion)" ] ; then \
-	    run_ldconfig=no ; rm -f "$(DESTDIR)$(libdir)/lib$(libname).so.$(soversion)" ; \
-	  else run_ldconfig=yes ; \
-	  fi ; \
-	  cd "$(DESTDIR)$(libdir)" && ln -s lib$(libname).so.$(pkgversion) lib$(libname).so.$(soversion) ; \
-	  if [ $${run_ldconfig} = yes ] ; then $(LDCONFIG) "$(DESTDIR)$(libdir)" ; fi ; \
+	  $(INSTALL_DATA) ./lib$(libname).dll.a "$(DESTDIR)$(libdir)/lib$(libname).dll.a" ; \
+	  $(INSTALL_PROGRAM) ./cyg$(libname)-$(soversion).dll "$(DESTDIR)$(bindir)/cyg$(libname)-$(soversion).dll" ; \
 	fi
 
 install-info :
