--- origsrc/lzlib-1.2/Makefile.in	2011-05-17 05:21:45.000000000 -0500
+++ src/lzlib-1.2/Makefile.in	2011-10-27 23:14:47.268808800 -0500
@@ -22,13 +22,13 @@ all : $(progname) $(progname_shared)
 lib$(libname).a: $(lib_objs)
 	$(AR) -rcs $@ $^
 
-lib$(libname).so.$(pkgversion) : $(sh_lib_objs)
-	$(CXX) -shared -Wl,--soname=lib$(libname).so.$(soversion) -o $@ $^
+lib$(libname).dll.a : $(sh_lib_objs)
+	$(CXX) -shared -Wl,--out-implib,$@ -o cyg$(libname)-$(soversion).dll $^
 
 $(progname) : $(objs) lib$(libname).a
 	$(CXX) $(LDFLAGS) -o $@ $^
 
-$(progname)_shared : $(objs) lib$(libname).so.$(pkgversion)
+$(progname)_shared : $(objs) lib$(libname).dll.a
 	$(CXX) $(LDFLAGS) -o $@ $^
 
 $(progname)_profiled : $(objs) lib$(libname).a
@@ -95,13 +95,9 @@ install : all install-info
 	$(INSTALL_DATA) $(VPATH)/$(libname)lib.h "$(DESTDIR)$(includedir)/$(libname)lib.h"
 	$(INSTALL_DATA) ./lib$(libname).a "$(DESTDIR)$(libdir)/lib$(libname).a"
 	if [ -n "$(progname_shared)" ] ; then \
-	  $(INSTALL_PROGRAM) ./lib$(libname).so.$(pkgversion) "$(DESTDIR)$(libdir)/lib$(libname).so.$(pkgversion)" ; \
-	  if [ -e "$(DESTDIR)$(libdir)/lib$(libname).so.$(soversion)" ] ; then \
-	    run_ldconfig=no ; rm -f "$(DESTDIR)$(libdir)/lib$(libname).so.$(soversion)" ; \
-	  else run_ldconfig=yes ; \
-	  fi ; \
-	  cd "$(DESTDIR)$(libdir)" && ln -s lib$(libname).so.$(pkgversion) lib$(libname).so.$(soversion) ; \
-	  if [ $${run_ldconfig} = yes ] && [ -x "$(LDCONFIG)" ] ; then "$(LDCONFIG)" -n "$(DESTDIR)$(libdir)" ; fi ; \
+	  if [ ! -d "$(DESTDIR)$(bindir)" ] ; then $(INSTALL_DIR) "$(DESTDIR)$(bindir)" ; fi ; \
+	  $(INSTALL_DATA) ./lib$(libname).dll.a "$(DESTDIR)$(libdir)/lib$(libname).dll.a" ; \
+	  $(INSTALL_PROGRAM) ./cyg$(libname)-$(soversion).dll "$(DESTDIR)$(bindir)/cyg$(libname)-$(soversion).dll" ; \
 	fi
 
 install-info :
